### DISCUSSION TITLE
Problem with auto focusing target when using DepthOfField

### DISCUSSION THREAD

**Discussion by maxoftime** [pmndrs/postprocessing#433] [Q&A]
Hi!
I really appreciate this library and the ease of use when creating nice effects like depth of field. I can't really wrap my head around the magic with setting an auto focus target. It just never focuses on the point I assign. In this example it should track (0,0,0) when activated. But instead it just focuses really close to the camera. 
https://codesandbox.io/s/inspiring-rain-dnl3ux?file=/app.js

Am I missing something obvious? 
Thanks in advance!

**Comment by vanruesc** [comment]
This might be a bug, but I'm not sure yet what causes it. Related: #426

**Reply by maxoftime** [reply]
It seems my issues stems from having logarithmicDepthBuffer set to true in my renderer. Which might make sense I guess (even though I'm not sure enough on how postprocessing handles depth).

However in my client case, that I can't share right now, postprocessing also had trouble finding any depth in my scene. Which then led to the bokeh applying to either everything or nothing. I'll try to recreate that error in a sharable sandbox. 

Thanks again for an awesome library, and also great help here in the discussion.

**Reply by vanruesc** [reply]
Thanks for the information! None of the depth-based effects currently support logarithmic depth. Does your client project also use log depth?

**Reply by maxoftime** [reply]
I think this scene is similar enough to act as a stand in to my client project. I experience the same issues at least. 
https://codesandbox.io/s/wonderful-williams-wfd4cu?file=/app.js

Without logarithmicDepthBuffer, depth works as intended (albeit a bit on the shallow side) if I set my camera.near to a value of 1 or above. When near is set to 0.1 (that I need) it gets unusable shallow. 

With logarithmicDepthBuffer active I get a nice spread across the scene, and reasonable worldFocusDistance values set using gui controls (cube in center is in focus at roughly far-value/2). The problem in this scenario is that I can't just assign the distance from camera to my target as a worldFocusDistance (since it's logarithmic).

I can't seem to get dofEffect.target to work in either case, but I suspect that issues because of my import of the library, using an importmap in index.html. I haven't looked in to this enough though.

TL;DR
* Camera near values below 1 seems to cause disturbances.
* LogarithmicDepthBuffer generates nice depth in scene but units are hard to calculate to match realWorldUnits needed for focus.
* Using importmaps seems to affect dofEffect.target. I'm thinking import and **this** struggles.
